crates/rune/src/languageserver/state.rs

```diff
@@ -250,6 +250,34 @@ impl<'a> State<'a> {
         Some(results)
     }
 
+    pub(super) fn format(&mut self, uri: &Url) -> Option<TextEdit> {
+        let sources = &mut self.workspace.sources;
+        tracing::trace!(uri = ?uri, uri_exists = sources.get(uri).is_some());
+        let workspace_source = sources.get_mut(uri)?;
+
+        // Not sure how to get source of `workspace_source`
+        let source = crate::source::Source::new(uri, workspace_source.content.to_string());
+        let formatted = crate::fmt::layout_source(&source).ok()?;
```

Comment:
Some kind of user-feedback when formatting errors would be nice too. AFAICT, this will no-op.

Reply:
if it doesn't fmt that would be a syntax error right? And a syntax error is reported anyway?

e.g. rust-analyzer also doesn't error when fmt fails. (I think only wen trailing whitespace remains)

Reply:
Maybe, then do what you feel is best.